File: Online Judges/LeetCode/June LeetCoding Challenge/Week-4/Single Number II.cpp
class Solution
{
public:
    int singleNumber(vector<int> &nums)
    {
        int first = 0, second = 0;
        for (auto num : nums)
        {
            first = (num ^ first) & ~second;
            second = (num ^ second) & ~first;
        }
        return first;
    }
};

File: Online Judges/LeetCode/June LeetCoding Challenge/Week-4/Single Number II.rs
impl Solution {
    pub fn single_number(nums: Vec<i32>) -> i32 {
        let mut first:i32 = 0;
        let mut second:i32 = 0;
        
        for num in nums{
            first = (first ^ num) & !second;
            second = (second ^ num) & !first;
        }
        first
    }
}